### PR TITLE
Update usage instructions

### PR DESCRIPTION
**Pull request by andreastt**
Expands the usage instructions section of the README to contain actual,
useful information on how to use geckodriver with Selenium and as a
standalone WebDriver server.

---
This change is [Reviewable](https://reviewable.io/reviews/mozilla/geckodriver/625)

### CODE REVIEW

**Comment by andreastt** [comment]
Rendered version at https://github.com/andreastt/geckodriver/tree/usage-instr#usage.

File: README.md [hunk 1]
```diff
@@ -323,13 +327,106 @@ and enable verbose logging:
...
+
+Or in the Windows console window:
+
+	$ set PATH=%PATH%;C:\bin\geckodriver
+
+### Standalone
+
+Since geckodriver is a separate HTTP server
+that is a complete remote end implementation of [WebDriver](https://w3c.github.io/webdriver/webdriver-spec.html),
+it is possible to circumvent the Selenium remote server
```
Comment (jgraham) [inline README.md]: I might say "it is possible to avoid using the Selenium remote server if you have no requirements to[…]"

File: README.md [hunk 2]
```diff
@@ -361,14 +458,18 @@ available in geckodriver by viewing the help message:
 
 ## Building
 
-geckodriver is written in [Rust](https://www.rust-lang.org/)
-and you need the [Rust toolchain](https://rustup.rs/) to compile it.
+geckodriver is written in [Rust](https://www.rust-lang.org/),
+a new systems programming language from the [Mozilla](https://www.mozilla.org/en-US/),
```
Comment (jgraham) [inline README.md]: s/new //

File: README.md [hunk 3]
```diff
@@ -361,14 +458,18 @@ available in geckodriver by viewing the help message:
...
+that is fast, prevents segfaults, and guarantees thread safety.
```
Comment (jgraham) [inline README.md]: So, it does prevent segfaults, but I think I would either not mention "why Rust" at all, or say: `fast, memory safe, and[…]`

File: README.md [hunk 4]
```diff
@@ -16,10 +16,10 @@ for a record of all notable changes to the program.
 are made available on GitHub
 on [supported platforms](#supported-firefoxen).
 
-## Supported Clients
-
-[Selenium](http://docs.seleniumhq.org/) users must update to [version 3.3.1](https://github.com/SeleniumHQ/selenium/releases/tag/selenium-3.3.1) or later to use geckodriver.
+## Supported clients
 
+[Selenium](http://docs.seleniumhq.org/) users
+must update to [version 3.3.1](https://github.com/SeleniumHQ/selenium/releases/tag/selenium-3.3.1)or later to use geckodriver.
```
Comment (thc202) [inline README.md]: It's missing a space after the link (`3.3.1or`).

File: README.md (hunk 1 above)
Comment (andreastt) [inline README.md]: Thanks, fixed.

File: README.md (hunk 2 above)
Comment (andreastt) [inline README.md]: Fixed.

File: README.md (hunk 3 above)
Comment (andreastt) [inline README.md]: I stole this directly from the website, but I’m OK to leave it out.

It now says: “geckodriver is written in Rust, a systems programming language from Mozilla. In order to build […]”

File: README.md (hunk 4 above)
Comment (andreastt) [inline README.md]: Fixed, thanks.

**Pull request merged**